test_replica_backup_and_restore random failure

**Issue Description**
CI test dirsrvtests/tests/suites/fourwaymmr/fourwaymmr_test.py::test_replica_backup_and_restore sometime fails during nighty tests::

```
        # Check that the updates (DEL) are no longer there
        for i in users.list():
            testuser = UserAccount(topo_m4.ms["supplier1"], i.dn)
>           assert testuser.exists()
E           assert False
E            +  where False = <bound method DSLdapObject.exists of <lib389.idm.user.UserAccount object at 0x7f6612653a60>>()
E            +    where <bound method DSLdapObject.exists of <lib389.idm.user.UserAccount object at 0x7f6612653a60>> = <lib389.idm.user.UserAccount object at 0x7f6612653a60>.exists

```
IMHO the issue is a simple timing issue in the test case:
  we should disable the agmt toward supplier1 before starting it 
  and re-enable them after checking that user exists

--- a/dirsrvtests/tests/suites/fourwaymmr/fourwaymmr_test.py
+++ b/dirsrvtests/tests/suites/fourwaymmr/fourwaymmr_test.py
@@ -430,6 +430,7 @@ def test_bob_acceptance_tests(topo_m4):
 
 
 def list_agmt_towards(topo_m4, serverid):
+    # Note: all instances must be started to use that.
     res = []
     for inst in topo_m4:
         for agmt in Agreements(inst).list():
@@ -484,6 +485,9 @@ def test_replica_backup_and_restore(topo_m4):
     for i in users.list(): topo_m4.ms["supplier1"].delete_s(i.dn)
     repl.wait_for_replication(topo_m4.ms["supplier1"], topo_m4.ms["supplier2"])
     repl.test_replication(topo_m4.ms["supplier1"], topo_m4.ms["supplier2"], 30)
+    # disable the agmt (while server is up) to avoid the DEL get replayed too early
+    for agmt in list_agmt_towards(topo_m4, "supplier1"):
+        agmt.pause()
     topo_m4.ms["supplier1"].stop()
     topo_m4.ms["supplier1"].ldif2db(
         bename=None,
@@ -492,9 +496,6 @@ def test_replica_backup_and_restore(topo_m4):
         suffixes=[DEFAULT_SUFFIX],
         import_file="/tmp/output_file",
     )
-    # disable the agmt to avoid the DEL get replayed too early
-    for agmt in list_agmt_towards(topo_m4, "supplier1"):
-        agmt.pause()
     topo_m4.ms["supplier1"].start()
 
     # Check that the updates (DEL) are no longer there

--- a/dirsrvtests/tests/suites/fourwaymmr/fourwaymmr_test.py
+++ b/dirsrvtests/tests/suites/fourwaymmr/fourwaymmr_test.py
@@ -429,6 +429,15 @@ def test_bob_acceptance_tests(topo_m4):
     repl.wait_for_replication(topo_m4.ms["supplier2"], topo_m4.ms["supplier1"])
 
 
+def list_agmt_towards(topo_m4, serverid):
+    res = []
+    for inst in topo_m4:
+        for agmt in Agreements(inst).list():
+            if agmt.get_attr_val_utf8(AGMT_PORT) == topo_m4.ms[serverid].port:
+                res.append(agmt)
+    return res
+
+
 @pytest.mark.bz830335
 def test_replica_backup_and_restore(topo_m4):
     """Test Backup and restore
@@ -483,12 +492,19 @@ def test_replica_backup_and_restore(topo_m4):
         suffixes=[DEFAULT_SUFFIX],
         import_file="/tmp/output_file",
     )
+    # disable the agmt to avoid the DEL get replayed too early
+    for agmt in list_agmt_towards(topo_m4, "supplier1"):
+        agmt.pause()
     topo_m4.ms["supplier1"].start()
 
     # Check that the updates (DEL) are no longer there
     for i in users.list():
         testuser = UserAccount(topo_m4.ms["supplier1"], i.dn)
         assert testuser.exists()
+
+    # Re enable the agmts
+    for agmt in list_agmt_towards(topo_m4, "supplier1"):
+        agmt.resume()
 
     # Here the changelog of supplier1 has been cleared.
     # Let's wait the supplier2 resync supplier1 BEFORE doing

